### PR TITLE
Support for conditional message sending

### LINKED ISSUE #1363: Process/forward messages conditionally based on twin state
Similar to the [conditional requests](https://www.eclipse.org/ditto/basic-conditional-requests.html) for CRUD of things it would be useful to apply a (RQL based) condition when processing messages in Ditto.

Example use cases:
* prevent sending duplicated alarm/alert messages from devices using a flag in the twin state:
   `POST /api/2/things/org.eclipse.ditto:my-thing-1/outbox/messages/fireAlarm?condition=ne(attributes/alarmConfirmed,true)`
   * e.g. a (mobile) app which would receive the "fireAlarm" for the first would set the `attributes/alarmConfirmed=true` 
   * this would prevent the mobile app to receive the fireAlarm multiple times
* only dispatch messages to a device when it is currently online:
   `POST /api/2/things/org.eclipse.ditto:my-thing-1/inbox/messages/doSomething?condition=gt(features/ConnectionStatus/properties/status/readyUntil,time:now)`

### PR DESCRIPTION
**Pull request by alstanchev**
- Only send messages to devices when a specified condition if fulfilled. 
- Ditto's HTTP Messages API supports to specify a "condition" query parameter with an RQL expression.
- Conditions are applied to the currently known twin/thing state
- When a condition is not passed, an immediate response is returned (e.g. HTTP 412 pre-condition failed) 

Fixes #1363

### CODE REVIEW

**Review by DerSchwilk — APPROVED** [review]
Reviewed as part of co-development

**Comment by thjaeckle** [comment]
@alstanchev for the blogpost you added, you will also need to add you to the authors file at https://github.com/eclipse-ditto/ditto/blob/master/documentation/src/main/resources/_data/authors.yml

**Comment by DerSchwilk** [comment]
Fixes https://github.com/eclipse-ditto/ditto/issues/1363

File: documentation/src/main/resources/_posts/2022-10-30-live-message-conditions.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+For this example, the authorized subject could have READ and WRITE permissions on the complete thing resource.
+However, it is only necessary on the path _thing:/features/alarm/properties/confirmed_.
+
+## Conditional requests via HTTP API
+Using the HTTP API the condition can either be specified via HTTP Header or via HTTP query parameter.  
+In this section, we will show how to use both options.
+
+### Conditional request with HTTP Header
+```
+curl -X PATCH -H 'Content-Type: application/json' -H 'condition: eq(features/alarm/properties/confirmed/,false)' /api/2/things/org.eclipse.ditto:carbon-monoxide-alarm/outbox/messages/co-alarm -d '{ "CO Level to high! Open your windows!" }'
```
Comment (thjaeckle) [inline documentation/src/main/resources/_posts/2022-10-30-live-message-conditions.md:105]: PATCH? Shouldn't sending a message use `POST`?

The host and port from the URL is also missing - did you try that example out?

File: documentation/src/main/resources/_posts/2022-10-30-live-message-conditions.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    },
+    "alarm": {
+      "properties": {
+        "lastTriggered": "2021-09-23T07:01:56Z",
+        "confirmed": false
+        }
+      }
+    }
+  }
+}
```
Comment (thjaeckle) [inline documentation/src/main/resources/_posts/2022-10-30-live-message-conditions.md:83]: the JSON is not valid .. one curly brace too much

**Comment by alstanchev** [comment]
@thjaeckle made some redactions on the blog post and also fixed the things that you mentioned above.

**Pull request merged**